### PR TITLE
#163519158 Implement capability of reporting an article

### PR DESCRIPTION
**Pull request by filschristian**
#### What does this PR do?
Add capability for authenticated user to report articles.

#### Description of Task to be completed?
this PR has the following end-point working:

- `POST: api/v1/articles/:slug/report`

- `GET: api/v1/articles/report`

#### How should this be manually tested?
Pull this branch

- Run `npm run migrate:seed`

- Run `npm run test`

- Run `npm run dev`

With postman send post requests to that API end-point

#### Any background context you want to provide?
This end-point are protected, they require authentication
Getting reports you must be an admin

#### What are the relevant pivotal tracker stories?
#163519158

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: controllers/ArticleController.js [hunk 1]
```diff
@@ -450,6 +450,68 @@ class ArticleController {
...
+   * @param {Object} res
+   * @param {*} next
+   * @returns {Object} Returns the response
+   */
+  static async reportArticle(req, res) {
+    const userId = req.currentUser.id;
+    const { slug } = req.params;
+    const { report } = req.body;
+
+    const article = await Article.findOne({ where: { slug } });
```
Comment (mcaleb808) [inline controllers/ArticleController.js]: @filschristian here you need also to check if the article is published

File: routes/api/articles.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ router.get(
   verifyJwt({ tokenRequired: false }),
   ArticleController.searchArticles
 );
+router.get('/report', verifyJwt({ tokenRequired: true }), ArticleController.getArticleReports);
```
Comment (mcaleb808) [inline routes/api/articles.js]: @filschristian Since you are not using try and catch in your controller, add asyncHandler to your router.

**Review by mcaleb808 — CHANGES_REQUESTED** [review]

File: controllers/ArticleController.js (hunk 1 above)
Comment (filschristian) [reply controllers/ArticleController.js]: Ahh thanks, Mr. Caleb. I will definitely check that

File: routes/api/articles.js (hunk 2 above)
Comment (filschristian) [reply routes/api/articles.js]: Good. Thanks. I will add it

File: __tests__/routes/rating.test.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  beforeAll(async () => {
+    const encryptedPassword = bcrypt.hashSync(signupUser.password, 10);
+    await User.create({
+      ...signupUser,
+      confirmed: 'confirmed',
+      password: encryptedPassword
+    });
+    const res = await request(app)
+      .post(`${urlPrefix}/users/login`)
+      .send({ user: { username: signupUser.email, password: signupUser.password } });
```
[the remark this answers is not in the document]
Comment (kimotho-njoki) [reply __tests__/routes/rating.test.js]: send({ 
     user: {
        username: signupUser.email, 
        password: signupUser.password 
}});

**Review by mcaleb808 — APPROVED** [review]

File: __tests__/routes/profile.test.js [hunk 4]
```diff
@@ -299,4 +299,4 @@ describe('Profile', () => {
     expect(res.body.message).toBe('The page does not exist');
     done();
   });
-});
+});
```
Comment (oesukam) [inline __tests__/routes/profile.test.js]: @filschristian Add an empty line after line 302

**Review by oesukam — CHANGES_REQUESTED** [review]

File: __tests__/routes/profile.test.js (hunk 4 above)
Comment (filschristian) [reply __tests__/routes/profile.test.js]: okay

**Review by oesukam — APPROVED** [review]

**Pull request merged**